tests/test_omxplayer.py

```diff
@@ -137,10 +137,113 @@ def run_command(self, command_name, *args):
+        self.player.stop()
+        self.assertEqual(self.callback_value, self.player)
+
+    def test_play_event(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player):
+            self.callback_value = player
+        
+        self.player.playEvent += callback
+        self.callback_value = None
+        self.player.play()
+        self.assertEqual(self.callback_value, self.player)
+
+    def test_play_event_by_play_pause(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player):
+            self.callback_value = player
+
+        self.assertEqual(self.player._is_playing, True)
+        # self.player._is_playing = True
+        self.player.pause()
+        self.player.playEvent += callback
+        self.callback_value = None
+        self.assertEqual(self.player._is_playing, False)
+        self.player.play_pause()
+        self.assertEqual(self.callback_value, self.player)
+
+    def test_pause_event(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player):
+            self.callback_value = player
+        
+        self.player.pauseEvent += callback
+        self.callback_value = None
+        self.player.pause()
+        self.assertEqual(self.callback_value, self.player)
+
+    def test_pause_event_by_play_pause(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player):
+            self.callback_value = player
+        
+        self.player.play() # make sure its playing
+        self.player.pauseEvent += callback
+        self.callback_value = None
+        self.player.play_pause() # toggle to pause
+        self.assertEqual(self.callback_value, self.player)
+
+    def test_seek_event(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player, position):
+            self.callback_value = (player, position)
+        
+        self.player.seekEvent += callback
+        self.callback_value = None
+        self.player.seek(3.4)
+        self.assertEqual(self.callback_value, (self.player, 3.4))
+
+    def test_position_event(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+
+        def callback(player, position):
+            self.callback_value = (player, position)
+        
+        self.player.positionEvent += callback
+        self.callback_value = None
+        self.player.set_position(5.01)
+        self.assertEqual(self.callback_value, (self.player, 5.01))
+
+    def test_load(self, popen, sleep, isfile, killpg, *args):
+        omxplayer_process = Mock()
+        popen.return_value = omxplayer_process
+        self.patch_and_run_omxplayer(active=True)
+        # initial load
+        self.assertEqual(self.player.get_filename(), './test.mp4')
+        killpg.assert_not_called()
+        popen.assert_called_once_with(
+            ['omxplayer', './test.mp4'],
+            preexec_fn=os.setsid, stdout=MOCK_OPEN())
```

Comment:
`os.setsid` is used to create a new process group which all children that are `fork`ed from a parent will inherit. When we kill `omxplayer.bin` we should be killing the process group id not the PID of `omxplayer.bin`. Currently we don't so I'm unsure `omxplayer.bin` reaps its children, thus leaving zombie processes.

I'm currently investigating the source of all the `dbus-daemon`s that are forked when we run the smoke tests, I initially thought it was `omxplayer.bin` but now I suspect it might be our library, I think `python-dbus` might be launching new daemons, but I haven't verified this yet.

`strace` can be used to track down what is going on.
`strace -f -e 'execv' ./test.py` run in `smoke_tests` will run the first smoke test and print out any `execv` calls (including children by use of the `f` flag), so we should be able to see the PID of the process spawning `dbus-daemon` and thus trace it back to either us or `omxplayer.bin`